### PR TITLE
correctly encode parameters

### PR DESCRIPTION
**Pull request by thehesiod**

### CODE REVIEW

File: aios3/bucket.py [hunk 1]
```diff
@@ -469,16 +468,26 @@ def __init__(self, bucket: Bucket):
...
-        elif '?versionId=' not in CopySource:
-            CopySource = amz_uriencode(CopySource)
-
-        response = await self._request("PUT", '/' + Key, 'CopyObject', headers={'x-amz-copy-source': CopySource})
+        # transform to dictionary
+        if not isinstance(CopySource, dict):
+            src_bucket, src_key = CopySource.split('/', 1)
+            CopySource = {'Bucket': src_bucket, 'Key': src_key}
+
+            if '?versionId=' in src_key:
```
Comment (EdRowe) [inline aios3/bucket.py:476]: What is the expected of the string version of CopySource? Is it a URL? This wouldn't be the normal way to parse a query string out of a URL

**Comment by thehesiod** [comment]
see http://botocore.readthedocs.io/en/latest/reference/services/s3.html#S3.Client.copy_object http://botocore.readthedocs.io/en/latest/reference/services/s3.html#S3.Client.copy_object
and http://docs.aws.amazon.com/AmazonS3/latest/API/RESTObjectCOPY.html http://docs.aws.amazon.com/AmazonS3/latest/API/RESTObjectCOPY.html

I actually tried it with string.printable however even botocore blows up, hah.  Have a simple testcase in the weather bucket backup unittest.

**Comment by FBNBuild** [comment]
Weird sorta-URL syntax but OK I guess that's how it works! ugh

**Comment by thehesiod** [comment]
their APIs leave much to be desired...and documented.  They just say the source needs to be url encoded, but they don't say which parts of the source :(

**Pull request merged**